android/src/main/java/com/codetrixstudio/capacitor/GoogleAuth/GoogleAuth.java

```diff
@@ -146,9 +136,57 @@ public void signOut(final PluginCall call) {
 
   @PluginMethod()
   public void initialize(final PluginCall call) {
+    String serverClientId = call.getString("serverClientId",
+                    this.getContext().getString(R.string.server_client_id));
+
+    boolean forceCodeForRefreshToken = call.getBoolean("forceCodeForRefreshToken", false);
+
+    GoogleSignInOptions.Builder googleSignInBuilder = initializeGoogleSignInBuilder(forceCodeForRefreshToken, serverClientId);
+
+    try {
+      JSArray scopeArray = call.getArray("scopes", new JSArray("[]"));
+      requestScopes(googleSignInBuilder, toStringArray(scopeArray));
+    } catch (JSONException e) {
+      Log.e("JSONException", e.getMessage());
+      call.reject(e.getMessage());
+    }
+
+    GoogleSignInOptions googleSignInOptions = googleSignInBuilder.build();
+    googleSignInClient = GoogleSignIn.getClient(this.getContext(), googleSignInOptions);
+
     call.resolve();
   }
 
+  private GoogleSignInOptions.Builder initializeGoogleSignInBuilder(boolean forceCodeForRefreshToken, String serverClientId) {
+    GoogleSignInOptions.Builder googleSignInBuilder = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+            .requestEmail();
+
+    // next step: rework this to actually be based on `grantOfflineAccess` + pass 'forceCode...' as 2nd param
+    if (forceCodeForRefreshToken) {
+      googleSignInBuilder.requestServerAuthCode(serverClientId, true);
+    } else {
+      googleSignInBuilder.requestIdToken(serverClientId);
+    }
```

Comment:
in the end I extracted `googleSignInBuilder` initialization to have the logic in a single place.

This if is based on this fragment of docs:
> You don't need to use [requestIdToken(String)](https://developers.google.com/android/reference/com/google/android/gms/auth/api/signin/GoogleSignInOptions.Builder#requestIdToken(java.lang.String)) when you use this option.

[src](https://developers.google.com/android/reference/com/google/android/gms/auth/api/signin/GoogleSignInOptions.Builder#public-googlesigninoptions.builder-requestserverauthcode-string-serverclientid)